docs/x509.rst

```diff
@@ -276,6 +292,24 @@ X.509 Certificate Object
 X.509 Extensions
 ~~~~~~~~~~~~~~~~
 
+.. class:: Extensions
+
+    .. versionadded:: 0.9
+
+    An X.509 Extensions instance is an ordered list of extensions.  The object
+    is iterable to get every extension.
+
+    .. method:: get_extension_for_oid(oid)
+
+        :param oid: An :class:`ObjectIdentifier` instance.
+
+        :returns: An instance of the extension class or None.
```

Comment:
Right now we return None, but I've talked a bit in the past about switching to an exception. It makes sense to me that if you explicitly request an extension and it does not exist maybe we should raise an exception (`x509.ExtensionNotFound` or something). It doesn't quite match the way we do `get_attributes_for_oid`, but that doesn't ever return none (only empty list).

Do you have a preference?